scene/resources/particles_material.cpp

```diff
@@ -187,6 +191,12 @@ void ParticlesMaterial::_update_shader() {
 				code += "uniform sampler2D emission_texture_color : hint_white;\n";
 			}
 		} break;
+		case EMISSION_SHAPE_RING: {
+			code += "uniform float ring_radius = 2.0;\n";
+			code += "uniform float ring_height = 1.0;\n";
+			code += "uniform float ring_inner_radius = 0.0;\n";
+			code += "uniform vec3 ring_axis = vec3(0.0,0.0,1.0);\n";
```

Comment:
It's tricky as `clang-format` obviously doesn't run on strings, but try to make the generated code follow the same style guide as the engine's `.glsl` (which is the same as for C++).
```suggestion
			code += "uniform vec3 ring_axis = vec3(0.0, 0.0, 1.0);\n";
```